tests/base_test.py

```diff
@@ -77,6 +95,20 @@ def test_config_property(union_value: typing.Union[int, str]):
     assert c.union == union_value
 
 
+def test_enum_config_property():
+    c1 = EnumTestConfig(enum='apple', enum_union='banana')
+    c2 = EnumTestConfig(enum='banana', enum_union='candy')
+    c3 = EnumTestConfig(enum='dragonfruit', enum_union='apple')
+    assert c1.enum == Enum1.apple
+    assert c2.enum == Enum1.banana
+    assert c1.enum_union == Enum1.banana
+    assert c2.enum_union == Enum2.candy
+    with raises(ConfigTypeError):
+        c3.enum
```

Comment:
Please check an error message as well. `ConfigTypeError` could be raised from many sources.